delivery_carrier_max_weight_constraint/wizards/choose_delivery_package.py

```diff
@@ -0,0 +1,25 @@
+# Copyright 2022 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import _, api, models
+from odoo.exceptions import ValidationError
+
+
+class ChooseDeliveryPackage(models.TransientModel):
+    _inherit = "choose.delivery.package"
+
+    @api.constrains("delivery_package_type_id", "shipping_weight")
+    def _check_strict_weight_package(self):
+        for package in self:
```

Comment:
@glitchov IMO,if `delivery_package_type_id` is not set, you should take the value for ` is_strict_weight`on the company.